File: public/section04/02.py
"""
1. 뒤집기
2. 소수 확인하기
"""

def solution(arr: list[int]) -> list[int]:
  q = []
  answer = []

  for num in arr:
    temp = num
    new = 0
    while True:
      q = temp // 10
      r = temp % 10
      new += r * 10

  return answer

print(solution())

File: public/section04/02.js
/**
 *
 */

function solution(arr = [1]) {
  let answer = [];
  for (let num of arr) {
    let s = num.toString();
    // let reversed = '';
    let reversed = s.split('').reverse().join('');

    // for (let i = 0; i < s.length; i++) {
    //   reversed += s[s.length - i - 1];
    // }

    let intNum = parseInt(reversed);
    if (intNum === 1) continue;

    let isPrime = true;
    for (let i = 2; i < Math.floor(intNum / 2); i++) {
      if (Math.floor(intNum % i) === 0) {
        isPrime = false;
        break;
      }
    }

    if (isPrime) {
      answer.push(intNum);
    }
  }

  return answer;
}

console.log(solution([32, 55, 62, 20, 250, 370, 200, 30, 100]));
